vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/create/AddProjectDialog.java

```diff
@@ -310,10 +350,49 @@ private void restrictInputLength() {
           e -> addConsumedLengthHelper(e.getSource(), e.getValue()));
     }
 
-    private void addConsumedLengthHelper(TextArea textArea, String newValue) {
-      int maxLength = textArea.getMaxLength();
-      int consumedLength = newValue.length();
-      textArea.setHelperText(consumedLength + "/" + maxLength);
+    private void generateProjectCode() {
+      generateCodeButton.addClickListener(
+          buttonClickEvent -> setCodeFieldValue(ProjectCode.random().value()));
+    }
+
+    private void resetDialogueUponClosure() {
+      // Calls the reset method for all possible closure methods of the dialogue window:
+      addDialogCloseActionListener(closeActionEvent -> resetAndClose());
+    }
+
+    private void closeDialogueViaCancelButton() {
+      cancelButton.addClickListener(buttonClickEvent -> resetAndClose());
+    }
+
+    private void configureValidators() {
+      Binder<Container<String>> binderTitle = new Binder<>();
+      binderTitle.forField(titleField)
+          .withValidator(value -> !value.isBlank(), "Please provide a title")
+          .bind(Container::value, Container::setValue);
+      Binder<Container<String>> binderObjective = new Binder<>();
+      binderObjective.forField(projectObjective)
+          .withValidator(value -> !value.isBlank(), "Please provide an " + "objective")
```

Comment:
AFAIK we also expect the species specimen and analyte to be set within this dialog and should validate accordingly?

Reply:
If that is not the case then the fields should not be set as required

Reply:
Hi @Steffengreiner , this is out of scope of this PR. Please open an issue in Jira, if you think this needs to be assessed.